src/include/migraphx/op/deconvolution.hpp

```diff
@@ -39,25 +39,27 @@ struct deconvolution
     std::string name() const { return "deconvolution"; }
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.has(2).same_type().same_ndims().only_dims(4);
+        check_shapes{inputs, *this}.has(2).same_type().same_ndims().min_ndims(3);
```

Comment:
since GPU has a max_ndims(5), do we need add here?

Reply:
the cpu implementation will be able to handle ndims > 5, but we restrict it for the gpu because of miopen

Reply:
I agree we shouldn't limit max dims to 5.